src/vm/eventpipe.cpp

```diff
@@ -125,37 +126,33 @@ void EventPipe::Shutdown()
         return;
     }
 
+    if (s_state != EventPipeState::Initialized)
+        return;
+
     // We are shutting down, so if disabling EventPipe throws, we need to move along anyway.
     EX_TRY
     {
-        bool tracingInitialized = false;
         {
             CrstHolder _crst(GetLock());
-            tracingInitialized = s_tracingInitialized;
```

Comment:
If we do have race conditions that create concurrent shutdown calls then checking the state under the lock is required for correctness. Otherwise a 2nd instance of Shutdown could start concurrently after the check at line 129 and before we've entered this lock.